### PR TITLE
SPCTR-1015:Fix:  Buttons gap settings on mobile preview gap settings is taking desktop value as default.

### PR DESCRIPTION
**Pull request by rajmohanreddy774**
### Description

Buttons gap settings on mobile preview gap settings is taking desktop value as default.

### Checklist:
- [ ] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

**Review by AbhijitNage123 — APPROVED** [review]

**Review by AbhijitNage123 — APPROVED** [review]

**Review by AbhijitNage123 — APPROVED** [review]

File: blocks-config/uagb-controls/generateCSS.js [hunk 1]
```diff
@@ -33,10 +33,24 @@ function generateCSS(
 		}
 
 		if ( css.length !== 0 ) {
-			gen_styling_css += id;
-			gen_styling_css += selector + '{';
-			gen_styling_css += css;
-			gen_styling_css += '}';
+			if ( responsiveType === 'tablet' ) {
```
Comment (supriyajjadhav) [inline blocks-config/uagb-controls/generateCSS.js:36]: @rajmohanreddy774 We have fixed the editor mobile issue in this [PR](https://github.com/brainstormforce/ultimate-addons-for-gutenberg/pull/2940) 
Can you please remove these changes from here? 
Once #2940 will merge in next-release then we will test and merge this PR. 
Till then Please keep this card blocked.

File: includes/blocks/buttons/frontend.css.php [hunk 2]
```diff
@@ -23,16 +23,19 @@
...
-		'row-gap' => UAGB_Helper::get_css_value( $attr['gapMobile'], 'px' ),
+		'flex-direction' => 'column',
+		'row-gap'        => UAGB_Helper::get_css_value( $attr['gapMobile'], 'px' ),
 	);
 
 } elseif ( 'tablet' === $attr['stack'] ) {
 
 	$selectors['.wp-block-uagb-buttons.uagb-buttons__outer-wrap  .uagb-buttons__wrap '] = array(
-		'column-gap' => UAGB_Helper::get_css_value( UAGB_Block_Helper::get_fallback_number( $attr['gap'], 'gap', $block_name ), 'px' ),
+		'flex-direction' => 'column',
```
Comment (supriyajjadhav) [inline includes/blocks/buttons/frontend.css.php:37]: if  'tablet' === $attr['stack'] then on desktop view `flex-direction` should be `row`. Please check - https://share.getcloudapp.com/Kou9E00X

File: includes/blocks/buttons/frontend.css.php [hunk 3]
```diff
@@ -46,10 +49,12 @@
 } elseif ( 'mobile' === $attr['stack'] ) {
 
 	$selectors['.wp-block-uagb-buttons.uagb-buttons__outer-wrap .uagb-buttons__wrap ']  = array(
-		'column-gap' => UAGB_Helper::get_css_value( UAGB_Block_Helper::get_fallback_number( $attr['gap'], 'gap', $block_name ), 'px' ),
+		'flex-direction' => 'column',
```
Comment (supriyajjadhav) [inline includes/blocks/buttons/frontend.css.php:52]: if  'mobile' === $attr['stack'] then on desktop and tablet view `flex-direction` should be `row`. Please check - https://share.getcloudapp.com/Kou9E00X

File: src/blocks/buttons/styling.js [hunk 4]
```diff
@@ -251,9 +236,9 @@ function styling( props ) {
 	const base_selector = ` .uagb-block-${ props.clientId.substr( 0, 8 ) }`;
 	let styling_css = generateCSS( selectors, base_selector );
 
-	styling_css += generateCSS( tabletSelectors, `${ base_selector }.uagb-editor-preview-mode-tablet`, true, 'tablet' );
+	styling_css += generateCSS( tabletSelectors, `${ base_selector }`, true, 'tablet' );
```
Comment (supriyajjadhav) [inline src/blocks/buttons/styling.js:239]: Remove this change as I have fixed it in #2940

File: blocks-config/uagb-controls/generateCSS.js (hunk 1 above)
Comment (rajmohanreddy774) [reply blocks-config/uagb-controls/generateCSS.js]: Yes this card is on hold till global fix card is merged

File: includes/blocks/buttons/frontend.css.php (hunk 2 above)
Comment (rajmohanreddy774) [reply includes/blocks/buttons/frontend.css.php]: fixed

File: src/blocks/buttons/styling.js (hunk 4 above)
Comment (rajmohanreddy774) [reply src/blocks/buttons/styling.js]: removed

**Review by RajkiranBagal — APPROVED** [review]

File: src/blocks/buttons/styling.js [hunk 5]
```diff
@@ -106,67 +106,52 @@ function styling( attributes, clientId, name ) {
...
-			'.uagb-editor-preview-mode-mobile .uagb-buttons-stack-desktop .block-editor-block-list__layout'
-		] = {
+		mobileSelectors[ '.uagb-buttons__outer-wrap .block-editor-block-list__layout' ] = {
 			'flex-direction': 'column',
 			'row-gap': generateCSSUnit( gapMobile, 'px' ),
 		};
 	} else if ( 'tablet' === stack ) {
-		selectors[ '.uagb-editor-preview-mode-desktop .uagb-buttons-stack-tablet .block-editor-block-list__layout' ] = {
+		selectors[ '.uagb-buttons__outer-wrap .block-editor-block-list__layout' ] = {
+			'flex-direction': 'row',
```
Comment (supriyajjadhav) [inline src/blocks/buttons/styling.js:123]: Please update to this -'flex-direction': 'column'

File: src/blocks/buttons/styling.js [hunk 6]
```diff
@@ -106,67 +106,52 @@ function styling( attributes, clientId, name ) {
...
-		] = {
-			'flex-direction': 'column',
-			'row-gap': generateCSSUnit( gapMobile, 'px' ),
+		mobileSelectors[ '.uagb-buttons__outer-wrap .block-editor-block-list__layout' ] = {
+			'flex-direction': 'row',
+			'column-gap': generateCSSUnit( gapMobile, 'px' ),
 		};
 	} else if ( 'mobile' === stack ) {
-		selectors[ '.uagb-editor-preview-mode-desktop .uagb-buttons-stack-mobile .block-editor-block-list__layout' ] = {
+		selectors[ '.uagb-buttons__outer-wrap .block-editor-block-list__layout' ] = {
```
Comment (supriyajjadhav) [inline src/blocks/buttons/styling.js:135]: Please update to this -'flex-direction': 'column'

File: src/blocks/buttons/styling.js [hunk 7]
```diff
@@ -106,67 +106,52 @@ function styling( attributes, clientId, name ) {
...
+			'flex-direction': 'row',
+			'column-gap': generateCSSUnit( gapTablet, 'px' ),
 		};
-		mobileSelectors[
-			'.uagb-editor-preview-mode-mobile .uagb-buttons-stack-tablet .block-editor-block-list__layout'
-		] = {
-			'flex-direction': 'column',
-			'row-gap': generateCSSUnit( gapMobile, 'px' ),
+		mobileSelectors[ '.uagb-buttons__outer-wrap .block-editor-block-list__layout' ] = {
+			'flex-direction': 'row',
```
Comment (supriyajjadhav) [inline src/blocks/buttons/styling.js:131]: Please update to this -'flex-direction': 'column'

File: includes/blocks/buttons/frontend.css.php [hunk 8]
```diff
@@ -18,52 +18,57 @@
...
-		'column-gap' => UAGB_Helper::get_css_value( UAGB_Block_Helper::get_fallback_number( $attr['gap'], 'gap', $block_name ), 'px' ),
+	$selectors['.uagb-buttons__outer-wrap .uagb-buttons__wrap']   = array(
+		'flex-direction' => 'row',
+		'column-gap'     => UAGB_Helper::get_css_value( UAGB_Block_Helper::get_fallback_number( $attr['gap'], 'gap', $block_name ), 'px' ),
 	);
-	$t_selectors[' .uagb-buttons__wrap'] = array(
-		'flex-direction' => 'column',
-		'row-gap'        => UAGB_Helper::get_css_value( $attr['gapTablet'], 'px' ),
+	$t_selectors['.uagb-buttons__outer-wrap .uagb-buttons__wrap'] = array(
+		'flex-direction' => 'row',
```
Comment (supriyajjadhav) [inline includes/blocks/buttons/frontend.css.php:41]: Please update to this -'flex-direction': 'column'

File: src/blocks/buttons/styling.js [hunk 9]
```diff
@@ -106,67 +106,52 @@ function styling( attributes, clientId, name ) {
...
 			'column-gap': generateCSSUnit( getFallbackNumber( gap, 'gap', blockName ), 'px' ),
-			'align-items': 'center',
 		};
-		tabletSelectors[
-			'.uagb-editor-preview-mode-tablet .uagb-buttons-stack-tablet .block-editor-block-list__layout'
-		] = {
-			'flex-direction': 'column',
-			'row-gap': generateCSSUnit( gapTablet, 'px' ),
+		tabletSelectors[ '.uagb-buttons__outer-wrap .block-editor-block-list__layout' ] = {
+			'flex-direction': 'row',
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply src/blocks/buttons/styling.js]: changed

File: src/blocks/buttons/styling.js (hunk 7 above)
Comment (rajmohanreddy774) [reply src/blocks/buttons/styling.js]: updated

File: src/blocks/buttons/styling.js [hunk 10]
```diff
@@ -106,67 +106,52 @@ function styling( attributes, clientId, name ) {
...
 			'column-gap': generateCSSUnit( gapTablet, 'px' ),
-			'align-items': 'center',
 		};
-		mobileSelectors[
-			'.uagb-editor-preview-mode-mobile .uagb-buttons-stack-mobile .block-editor-block-list__layout'
-		] = {
-			'flex-direction': 'column',
-			'row-gap': generateCSSUnit( gapMobile, 'px' ),
+		mobileSelectors[ '.uagb-buttons__outer-wrap .block-editor-block-list__layout' ] = {
+			'flex-direction': 'row',
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply src/blocks/buttons/styling.js]: updated

File: includes/blocks/buttons/frontend.css.php (hunk 8 above)
Comment (rajmohanreddy774) [reply includes/blocks/buttons/frontend.css.php]: updated

File: includes/blocks/buttons/frontend.css.php [hunk 11]
```diff
@@ -18,52 +18,57 @@
...
+		'column-gap'     => UAGB_Helper::get_css_value( $attr['gapTablet'], 'px' ),
 	);
-	$m_selectors[' .uagb-buttons__wrap'] = array(
-		'flex-direction' => 'column',
-		'row-gap'        => UAGB_Helper::get_css_value( $attr['gapMobile'], 'px' ),
+	$m_selectors['.uagb-buttons__outer-wrap .uagb-buttons__wrap'] = array(
+		'flex-direction' => 'row',
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply includes/blocks/buttons/frontend.css.php]: updated

File: includes/blocks/buttons/frontend.css.php [hunk 12]
```diff
@@ -18,52 +18,57 @@
...
-		'column-gap' => UAGB_Helper::get_css_value( $attr['gapTablet'], 'px' ),
+	$t_selectors['.uagb-buttons__outer-wrap .uagb-buttons__wrap'] = array(
+		'flex-direction' => 'row',
+		'column-gap'     => UAGB_Helper::get_css_value( $attr['gapTablet'], 'px' ),
 	);
-	$m_selectors[' .uagb-buttons__wrap'] = array(
-		'flex-direction' => 'column',
-		'row-gap'        => UAGB_Helper::get_css_value( $attr['gapMobile'], 'px' ),
+	$m_selectors['.uagb-buttons__outer-wrap .uagb-buttons__wrap'] = array(
+		'flex-direction' => 'row',
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply includes/blocks/buttons/frontend.css.php]: updated

**Review by supriyajjadhav — APPROVED** [review]

File: src/blocks/buttons/styling.js [hunk 13]
```diff
@@ -106,67 +106,63 @@ function styling( attributes, clientId, name ) {
...
+		mobileSelectors[ '.uagb-buttons__outer-wrap .block-editor-block-list__layout' ] = {
 			'flex-direction': 'column',
 			'row-gap': generateCSSUnit( gapMobile, 'px' ),
 		};
 	} else if ( 'tablet' === stack ) {
-		selectors[ '.uagb-editor-preview-mode-desktop .uagb-buttons-stack-tablet .block-editor-block-list__layout' ] = {
+		selectors[ '.uagb-buttons__outer-wrap .block-editor-block-list__layout' ] = {
+			'flex-direction': 'row',
 			'column-gap': generateCSSUnit( getFallbackNumber( gap, 'gap', blockName ), 'px' ),
-			'align-items': 'center',
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply src/blocks/buttons/styling.js:130]: It was there in the previous dev code, i think it is of no use I will remove it.

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**